internal/kafka/internal/handlers/admin_kafka.go

```diff
@@ -169,13 +171,33 @@ func (h *adminKafkaHandler) Update(w http.ResponseWriter, r *http.Request) {
 				return false
 			}
 
+			getStatusBasedOnSuspendedParam := func(susp *bool, kafka *dbapi.KafkaRequest) string {
+				if shared.IsNilPredicate(susp) {
+					return kafka.Status
+				} else {
+					if *susp {
+						if kafka.Status != constants.KafkaRequestStatusSuspended.String() {
+							return constants.KafkaRequestStatusSuspending.String()
+						}
+					} else {
+						if kafka.Status == constants.KafkaRequestStatusSuspended.String() || kafka.Status == constants.KafkaRequestStatusSuspending.String() {
```

Comment:
We probably also want to add in `KafkaRequestStatusFailed` to this check so users can't update from `failed` -> `resuming`

Reply:
read this wrong, ignore this :)